pkg/cloud-provider/cloudapi/azure/azure_services.go

```diff
@@ -54,15 +53,16 @@ type azureServicesHelperImpl struct{}
 // newServiceSdkConfigProvider returns config to create azure services clients.
 func (h *azureServicesHelperImpl) newServiceSdkConfigProvider(accCreds *azureAccountConfig) (
 	azureServiceClientCreateInterface, error) {
-	var authorizer autorest.Authorizer
 	var err error
-	clientConfig := auth.NewClientCredentialsConfig(accCreds.ClientID, accCreds.ClientKey, accCreds.TenantID)
-	authorizer, err = clientConfig.Authorizer()
+
+	//TODO: Expose an option in CPA to specify the cloud type, AzurePublic, AzureGovernment and AzureChina.
+	cred, err := azidentity.NewClientSecretCredential(accCreds.TenantID, accCreds.ClientID, accCreds.ClientKey, nil)
 	if err != nil {
 		return nil, fmt.Errorf("unable to initialize Azure authorizer from credentials: %v", err)
```

Comment:
can you simulate this and see if CPA is updated with error?

Reply:
No, we do not update CPA with error. Any error in newCloudAccountConfi()  is getting propagated to the  CPA reconciler, hence I think it is a good idea to update CPA status with Error in CloudproviderAccount_controller.go. Do you have any other thoughts here?

Reply:
When tried with invalid credentials(invalid clientid or invalid tenant id), cred creation was successful, hence there weren't any errors from AddProviderAccount() call in CPA reconciler. While making a cloud call for getting vpcs, request errors out and we update CPA CR with the error .